src/items/external-blocks.md

```diff
@@ -186,3 +205,4 @@ restrictions as [regular function parameters].
 [_Visibility_]: ../visibility-and-privacy.md
 [attributes]: ../attributes.md
 [regular function parameters]: functions.md#attributes-on-function-parameters
+[documentation for rustc]: https://doc.rust-lang.org/rustc/command-line-arguments.html#-l-link-the-generated-crate-to-a-native-library
```

Comment:
We generally prefer relative links to other books that are published by rust-lang. This helps with offline viewing, and keeping the user within the same release when viewing release-specific URLs like https://doc.rust-lang.org/1.59.0/reference/. Our linkchecker also only supports relative links, and this helps prevent links from going stale. This would also avoid confusion, as the existing link doesn't have any information on whole-archive which can be a bit confusing for someone looking at the nightly docs.

Also, I feel like it would be a little clearer if it linked directly to the whole-archive section, unless you had some specific reason to link to the `-l` section?

```suggestion
[documentation for rustc]: ../../rustc/command-line-arguments.html#linking-modifiers-whole-archive
```

Reply:
Looks like linkchecker doesn't like it though.

Reply:
Ah, ok, external links should indeed use html, not md.